src/foam/java/Enum.js

```diff
@@ -1,73 +1,55 @@
-// Test with: foam.u2.ControllerMode.model_.javaSource();
+// Defines template for the EnumModel
```

Comment:
Overall I wonder if we can do some better code sharing by making a foam.java.Enum class, and having it extend foam.java.Class

An Enum is in essence a Class with a fixed number of implementations.  If we extend foam.java.Class and update the outputJava method to output the VALUES, as well as whatever tweaks we need to match java enum syntax, we should also get methods/fields for free.


I would start by defining foam.java.Enum as a foam.CLASS() that extends foam.java.Class, and then trying to get the output correct.  Then refine buildJavaClass() onto foam.core.EnumModel, which should build a foam.java.Enum.